api/apikeys.go

```diff
@@ -0,0 +1,69 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/SkynetLabs/skynet-accounts/jwt"
+	"github.com/julienschmidt/httprouter"
+)
+
+// userAPIKeyPOST creates a new API key for the user.
+func (api *API) userAPIKeyPOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := jwt.TokenFromContext(req.Context())
+	if err != nil {
+		api.WriteError(w, err, http.StatusUnauthorized)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub, true)
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	ak, err := api.staticDB.APIKeyCreate(req.Context(), *u)
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	api.WriteJSON(w, ak)
+}
+
+// userAPIKeyGET lists all API keys associated with the user.
+func (api *API) userAPIKeyGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := jwt.TokenFromContext(req.Context())
+	if err != nil {
+		api.WriteError(w, err, http.StatusUnauthorized)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub, true)
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	aks, err := api.staticDB.APIKeyList(req.Context(), *u)
```

Comment:
Referring to my previous comment. We might not actually want to return the actual keys here but only some information like a friendly name or id, when it was created, when it was used the last time etc. Basically information that tells a user they have keys and a way to identify them but actually the secret.

Neither Github nor Gitlab seem to allow you to fetch a token after its creation so it might be good practice now to not hand them out after creation.

Reply:
Good point. We shouldn't display them more than once. I'll change the list endpoint to return only the key's ID and time of creation and the delete endpoint to take a key ID.